kdc/default_config.c

```diff
@@ -59,6 +59,7 @@ krb5_kdc_get_config(krb5_context context, krb5_kdc_configuration **config)
     c->check_ticket_addresses = TRUE;
     c->allow_null_ticket_addresses = TRUE;
     c->allow_anonymous = FALSE;
+    c->historical_anon_realm = FALSE;
```

Comment:
I think that we should consider making the default true in the 7.x releases. This is a breaking change for Heimdal shops and such changes should be restricted to major releases.

Reply:
I defer to @jaltman on this one. Arguably the previous behavior is a security issue, as it reveals the realm for anonymous PKINIT.

Reply:
I agree with @lhoward.  The combination of anonymous PKINIT plus cross-realm results in a loss of privacy which is a security issue.   We should belatedly create a CVE describing the prior behavior.

Reply:
Right, but this potentially actually breaks existing installations which may have come to depend on the prior behaviour. This kind of change should generally be reserved for a major release or a default off option.

I don't think that the severity of the issue is severe enough to break things. It's a small reduction in anonymity only. I haven't properly checked, but can't you just back track the transited path and see where they came from in any case? In a cursory glance at the code, it appears that your client's realm will be added to the path because crealm != WELLKONWN:ANONYMOUS.

Reply:
I don't even see it as a reduction in anonymity, since the real realm will be in the transit path.

Reply:
Hmm, well yes potentially adding the anonymous client's realm is a bug in the KDC.

Reply:
I've not yet restored the historical default, will chat with Roland and Nico today or tomorrow.

Reply:
IMO the loss of privacy is not CVE-worthy.  Does anyone use Kerberos, with Heimdal 7.x, in such a way that they have cross-realm trusts and apps where there's a privacy concern?  I don't believe anyone is.